jinja2/environment.py

```diff
@@ -769,15 +769,7 @@ def join_path(self, template, parent):
     def _load_template(self, name, globals):
         if self.loader is None:
             raise TypeError('no loader for this environment specified')
-        try:
-            # use abs path for cache key
-            cache_key = self.loader.get_source(self, name)[1]
-        except RuntimeError:
-            # if loader does not implement get_source()
-            cache_key = None
-        # if template is not file, use name for cache key
-        if cache_key is None:
-            cache_key = name
+        cache_key = "%d%s" % (hash(self.loader), name)
```

Comment:
What about `cache_key = (self.loader, name)`?

This avoids string conversion and hash collision.